### ISSUE TITLE
Doesn't work with scikit-learn>=0.23.0 as logsumexp was removed from sklearn.utils.fixes.

### ISSUE THREAD

**Issue by filaPro** [scikit-learn-contrib/metric-learn#289]
#### Description
Doesn't work with scikit-learn>=0.23.0 as logsumexp was removed from sklearn.utils.fixes.

#### Steps/Code to Reproduce

from metric_learn import NCA

#### Actual Results

cannot import name 'logsumexp' from 'sklearn.utils.fixes'

**Comment by perimosocordiae** [comment]
Looks like we're using the scikit-learn backport of `logsumexp`, which they removed in favor of using the official scipy version: https://github.com/scikit-learn/scikit-learn/pull/16725
